### PR TITLE
ci: Fixes to codecov configuration

### PR DESCRIPTION
**Pull request by maurelian**
**Description**

Fixes some issues in the current codecov config.

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -1,20 +1,32 @@
...
+  behavior: default
+  require_changes: true    # only post the comment if coverage changes
 ignore:
   - "l2geth"
   - "**/*.t.sol"
   - "op-bindings/bindings/*.go"
-  - "packages/contracts-bedrock/contracts/(vendor|libraries)/(trie|rlp)"
+  - "packages/contracts-bedrock/contracts/vendor"
+  - "packages/contracts-bedrock/contracts/libraries/trie"
+  - "packages/contracts-bedrock/contracts/libraries/rlp"
```
Comment (maurelian) [inline codecov.yml:11]: AFAICT the LHS is correct, but it seems to have been causing everything in `packages/contracts-bedrock/contracts/` to be ignored.

File: codecov.yml [hunk 2]
```diff
@@ -1,20 +1,32 @@
-comment: true
+comment:
+  layout: "reach, diff, flags, files"
+  behavior: default
+  require_changes: true    # only post the comment if coverage changes
```
Comment (maurelian) [inline codecov.yml:4]: Hopefully this will actually enable the bot.

**Review by zhwrd — APPROVED** [review]

File: codecov.yml (hunk 1 above)
Comment (tynes) [inline codecov.yml:11]: I believe we have test coverage of `rlp`

Comment (tynes) [reply codecov.yml:11]: And differential test coverage of `trie`

**Review by tynes — APPROVED** [review]

File: codecov.yml [hunk 3]
```diff
@@ -1,20 +1,32 @@
...
 ignore:
   - "l2geth"
   - "**/*.t.sol"
   - "op-bindings/bindings/*.go"
-  - "packages/contracts-bedrock/contracts/(vendor|libraries)/(trie|rlp)"
+  - "packages/contracts-bedrock/contracts/vendor"
```
[the remark this answers is not in the document]
Comment (maurelian) [reply codecov.yml]: Good call

**Pull request merged**